From 9dde2ed5ea898f5cd1595924e1832ecad51e3e3d Mon Sep 17 00:00:00 2001
From: yaravapa <yasoda.aravapalli@intel.com>
Date: Wed, 13 Oct 2021 23:21:25 +0530
Subject: [PATCH] Avx enablment of expf, sinf, cosf

This patch enables avx2 verions of libm functions
sinf, cosf and expf.

Tracked-On: OAM-96244
Signed-off-by: yaravapa <yasoda.aravapalli@intel.com>
---
 Android.bp  | 11 +++++++++++
 math/expf.c |  5 +++++
 2 files changed, 16 insertions(+)

diff --git a/Android.bp b/Android.bp
index ea477a1..ad505a4 100755
--- a/Android.bp
+++ b/Android.bp
@@ -43,6 +43,17 @@ cc_library {
                 "-DHAVE_FAST_FMA=1",
             ],
         },
+        x86_64: {
+           // This flag is required to enable __math_may_uflowf
+           // when compiling expf from libm for x86_64 target
+           cflags: ["-DWANT_ERRNO_UFLOW"],
+           exclude_srcs: [
+               "math/sinf.c",
+               "math/cosf.c",
+               "math/expf.c",
+           ],
+        },
+
     },
 
     target: {
diff --git a/math/expf.c b/math/expf.c
index 0fe1f7d..d0be055 100644
--- a/math/expf.c
+++ b/math/expf.c
@@ -86,6 +86,11 @@ expf (float x)
   return eval_as_float (y);
 }
 #if USE_GLIBC_ABI
+#ifdef FROM_LIBM
+strong_alias (expf_generic, __expf_finite)
+hidden_alias (expf_generic, __ieee754_expf)
+#else
 strong_alias (expf, __expf_finite)
 hidden_alias (expf, __ieee754_expf)
 #endif
+#endif
-- 
2.17.1

